src/main/java/org/opentripplanner/routing/stoptimes/StopTimesHelper.java

```diff
@@ -44,26 +48,38 @@
       long startTime,
       int timeRange,
       int numberOfDepartures,
-      boolean omitNonPickups,
+      ArrivalDeparture arrivalDeparture,
       boolean includeCancelledTrips
   ) {
-
-    //
-    // TODO: SIRI: Add support for including cancelled trips
-    //
-
-
     if (startTime == 0) {
       startTime = System.currentTimeMillis() / 1000;
     }
-    List<StopTimesInPattern> ret = new ArrayList<>();
+    List<StopTimesInPattern> result = new ArrayList<>();
     Date date = new Date(startTime * 1000);
     ServiceDate[] serviceDates = {new ServiceDate(date).previous(), new ServiceDate(date), new ServiceDate(date).next()};
 
-    Collection<TripPattern> patternsForStop = routingService.getPatternsForStop(stop, true);
+    Collection<TripPattern> plannedPatterns = routingService.getPatternsForStop(stop, false);
+    Collection<TripPattern> realTimePatterns = routingService.getPatternsForStop(stop, true);
 
-    for (TripPattern pattern : patternsForStop) {
+    // Only include the planned patterns that are not also part of the collection of realtime
+    // patterns. These may have been cancelled with no replacement.
+    plannedPatterns.removeAll(realTimePatterns);
+
+    // Realtime patterns may have been cancelled and replaced. Do not include the patterns that
+    // have been replaced
+    Collection<TripPattern> replacedPatterns = realTimePatterns
```

Comment:
Can be replaced with built-in method `realtimePatterns.removeIf(tp -> tp.getOriginalTripPattern() != null)`

Reply:
I missed a detail during the discussion. Gard looked into this:  "it is not so easy to replace the removal logic with removeIf. It doesn’t remove the patterns that have an original pattern reference, but all the patterns that are referred to as original patterns by other patterns."